File: packages/app/ios/Plugin/FirebaseAppPlugin.swift
import Foundation
import Capacitor

import FirebaseCore

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(FirebaseAppPlugin)
public class FirebaseAppPlugin: CAPPlugin {

    override public func load() {
        if FirebaseApp.app() == nil {
            FirebaseApp.configure()
        }
    }

    @objc func getName(_ call: CAPPluginCall) {
        call.resolve([
            "name": FirebaseApp.app()?.name ?? ""
        ])
    }

    @objc func getOptions(_ call: CAPPluginCall) {
        call.resolve([
            "apiKey": FirebaseApp.app()?.options.apiKey ?? "",
            "applicationId": FirebaseApp.app()?.options.googleAppID ?? "",
            "databaseUrl": FirebaseApp.app()?.options.databaseURL ?? "",
            "gcmSenderId": FirebaseApp.app()?.options.gcmSenderID ?? "",
            "projectId": FirebaseApp.app()?.options.projectID ?? "",
            "storageBucket": FirebaseApp.app()?.options.storageBucket ?? ""
        ])
    }

    @objc func initializeApp(_ call: CAPPluginCall) {
        let name = call.getString("name")!
        let config = call.getObject("config")!
        let options = FirebaseOptions.init(googleAppID: config["appId"] as! String, gcmSenderID: config["messagingSenderId"] as! String)
        options.clientID = config["clientId"] as? String
        options.apiKey = config["apiKey"] as? String
        options.projectID = config["projectId"] as? String
        options.storageBucket = config["storageBucket"] as? String
        options.databaseURL = config["databaseURL"] as? String

        do {
            try FirebaseApp.configure(name: name, options: options)
            call.resolve()
        } catch {
            call.reject("Could not initialize app with provided Firebase config")
        }
    }

    @objc func getApps(_ call: CAPPluginCall) {
        let apps = FirebaseApp.allApps!
        call.resolve(createGetAppsResult(apps: apps))
    }

    public func createGetAppsResult(apps: [String: FirebaseApp]) -> JSObject {
        var result = JSObject()
        var appsArray = JSArray()
        for (name, _) in apps {
            appsArray.append(name)
        }
        result["apps"] = appsArray
        return result
    }

}

File: packages/app/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/capawesome/FirebaseAppPlugin.java
package io.capawesome.capacitorjs.plugins.firebase.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Logger;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@CapacitorPlugin(name = "FirebaseApp")
public class FirebaseAppPlugin extends Plugin {

    public static final String TAG = "FirebaseApp";

    @PluginMethod
    public void getName(PluginCall call) {
        try {
            String name = getFirebaseAppInstance().getName();

            JSObject result = new JSObject();
            result.put("name", name);
            call.resolve(result);
        } catch (Exception exception) {
            Logger.error(TAG, exception.getMessage(), exception);
            call.reject(exception.getMessage());
        }
    }

    @PluginMethod
    public void getOptions(PluginCall call) {
        try {
            FirebaseOptions options = getFirebaseAppInstance().getOptions();

            JSObject result = new JSObject();
            result.put("apiKey", options.getApiKey());
            result.put("applicationId", options.getApplicationId());
            result.put("databaseUrl", options.getDatabaseUrl());
            result.put("gcmSenderId", options.getGcmSenderId());
            result.put("projectId", options.getProjectId());
            result.put("storageBucket", options.getStorageBucket());
            call.resolve(result);
        } catch (Exception exception) {
            Logger.error(TAG, exception.getMessage(), exception);
            call.reject(exception.getMessage());
        }
    }

    private FirebaseApp getFirebaseAppInstance() {
        return FirebaseApp.getInstance();
    }

    @PluginMethod
    public void initializeApp(PluginCall call) {
        String name = call.getString("name");
        JSObject config = call.getObject("config");

        FirebaseOptions.Builder options = new FirebaseOptions.Builder();
        options
            .setApplicationId(Objects.requireNonNull(config.getString("appId")))
            .setApiKey(Objects.requireNonNull(config.getString("apiKey")))
            .setProjectId(Objects.requireNonNull(config.getString("projectId")))
            .setStorageBucket(Objects.requireNonNull(config.getString("storageBucket")))
            .setDatabaseUrl(Objects.requireNonNull(config.getString("databaseURL")))
            .setGcmSenderId(Objects.requireNonNull(config.getString("messagingSenderId")));

        try {
            FirebaseApp app = FirebaseApp.initializeApp(this.getContext(), options.build(), name);
            call.resolve();
        } catch (Exception error) {
            call.reject("Could not initialize app with provided firebase config: " + error.getLocalizedMessage());
        }
    }

    @PluginMethod
    public void getApps(PluginCall call) {
        String name = call.getString("name");
        List<FirebaseApp> apps = FirebaseApp.getApps(this.getContext());

        List<String> appNames = new ArrayList<>();

        for (FirebaseApp app : apps) {
            appNames.add(app.getName());
        }
        call.resolve(new JSObject().put("apps", appNames));
    }
}
